src/Types.php

```diff
@@ -36,58 +76,25 @@ class Types
      */
     public static function getPrimitiveTypeOf($instance)
     {
-        switch ($type = gettype($instance)) {
-            case 'array':
-                return self::TYPE_ARRAY;
-            case 'boolean':
-                return self::TYPE_BOOLEAN;
-            case 'integer':
-                return self::TYPE_INTEGER;
-            case 'NULL':
-                return self::TYPE_NULL;
-            case 'double':
-                return self::TYPE_NUMBER;
-            case 'object':
-                return self::TYPE_OBJECT;
-            case 'string':
-                return self::TYPE_STRING;
-        }
+        $phpType = gettype($instance);
 
-        throw new UnsupportedTypeException($type);
-    }
-
-    /**
-     * Returns whether an instance matches a given type.
-     *
-     * @param mixed  $instance
-     * @param string $type
-     *
-     * @return bool
-     */
-    public static function isA($instance, $type)
```

Comment:
What about restoring this method and make it use the `$phpJsonCompatibility` map internally? Would that hurt performance?